### PR TITLE
Fix no_upscale: Check that both dimensions are bigger than the original

### PR DESCRIPTION
**Pull request by tzi**
Hi @scalableminds!

There was an error with the condition that determine if we are in a upscale case or not.
If both dimensions are entered and only one dimension is bigger than the original, it was considered as an upscale case.

This false positive create a weird situation, where:
 * `no_upscale: false` => no resizing
 * `no_upscale: true` => downscale resizing

Thanks for sharing this lib!

Cheers,
Thomas.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,13 +30,6 @@ module.exports = function imageResizer(options) {
     format      : null
   });
 
-  if (options.height == null && options.width) {
-    options.height = null;
-  }
-  if (options.width == null && options.height) {
-    options.width = null;
-  }
-
```
Comment (tzi) [inline index.js]: Not related to this PR, but this code is not explicit and the tests pass without it.

File: index.js [hunk 2]
```diff
@@ -54,8 +47,8 @@ module.exports = function imageResizer(options) {
         if (options.height != null || options.width != null) {
 
           var isUpscaled =
-            (options.width && size.width < options.width) ||
-            (options.height && size.height < options.height);
```
Comment (tzi) [inline index.js]: Main part of this PR

File: index.js [hunk 3]
```diff
@@ -98,7 +91,7 @@ module.exports = function imageResizer(options) {
         }
 
         if (options.sharpen) {
-          gmfile = gmfile.unsharp(typeof options.sharpen=='string' ?  options.sharpen : '1.5x1+0.7+0.02');
```
Comment (tzi) [inline index.js]: jshint fix

File: index.js [hunk 4]
```diff
@@ -8,14 +8,7 @@
...
 var gm          = require("gulp-gm");
 var async       = require("async");
-var path        = require("path");
-var os          = require("os");
 var _           = require("lodash");
-var stream      = require("stream");
-var util        = require("gulp-util");
-var PluginError = util.PluginError;
-
-const PLUGIN_NAME = "gulp-image-resize";
```
Comment (tzi) [inline index.js]: Remove unused variables.

**Comment by normanrz** [comment]
Hey Thomas! Thanks for the PR. 
I just released a new version with your fixes. Sorry for the wait.

**Comment by tzi** [comment]
Thanks for the release!

Cheers,
Thomas.

**Pull request merged**